lib/yaml_permitted_classes.rb

```diff
@@ -0,0 +1,29 @@
+class YamlPermittedClasses
+  DEFAULT_PERMITTED_CLASSES = [Object, Range, Regexp, Symbol, Date, Time, DateTime, ActiveSupport::Duration, ActiveSupport::HashWithIndifferentAccess, ActiveSupport::TimeWithZone, ActiveSupport::TimeZone]
+  def self.app_yaml_permitted_classes
+    @app_yaml_permitted_classes ||= DEFAULT_PERMITTED_CLASSES + [MiqExpression]
+  end
+
+  def self.app_yaml_permitted_classes=(classes)
+    @app_yaml_permitted_classes = Array(classes)
+  end
```

Comment:
This is the mechanism added so plugins can enumerate their own additional permitted classes either in test or production.

From the commit:

For example, we can do this in tests:
```ruby
RSpec.configure do |config|
  config.before do
    YamlPermittedClasses.app_yaml_permitted_classes |= [VimHash, VimString, VimArray]
  end
end
```

Reply:
For production code, we may want a direct, well-defined, plugin method that we call, similar to how we set up loggers.  For example, in a plugin engine we can have something like 

```ruby
module ManageIQ::Providers::Vmware
  class Engine < ::Rails::Engine
    def yaml_permitted_classes
      [VimHash, VimString, VimArray]
    end
  end
end
```

Then in app initialization (probably in `Vmdb::Plugins.init`), we simply loop through all plugins that have that method implemented and set it up all at once.  Something like:

```ruby
module Vmdb::Plugins
  def init_yaml_permitted_classes
      each do |engine|
        classes = engine.try(:yaml_permitted_classes)
        YamlPermittedClasses.app_yaml_permitted_classes |= classes if classes.present?
      end
  end
```

If we do this, then in test don't think there's anything special to do, because plugins are initialized as part of the setup.